tests/Unit/Subscriber/Tools/TestDetectMissingTags.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace WP_Rocket\Tests\Unit\Subscriber\Tools;
+
+use WP_Rocket\Subscriber\Tools\Detect_Missing_Tags_Subscriber;
+use WP_Rocket\Tests\Unit\TestCase;
+use Brain\Monkey\Functions;
+
+class TestDetectMissingTags extends TestCase {
+
+	use \Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->mockCommonWpFunctions();
+
+		if ( ! defined('HOUR_IN_SECONDS') ) {
+			define('HOUR_IN_SECONDS', 60 * 60);
+		}
+	}
+
+	/**
+	 * Test Detect_Missing_Tags_Subscriber->maybe_missing_tags() with missing HTML code </html> and </body> are missing.
+	 */
+	public function testShouldIdentifyMissingHtmlAndBodyAndWPFooter() {
+		$missing_tag = new Detect_Missing_Tags_Subscriber();
+
+		$html = \file_get_contents( WP_ROCKET_PLUGIN_TESTS_ROOT . '/../Fixtures/Subscriber/Tools/original_no_html_and_body.html');
+
+		// Called did_action('wp_footer'), test also for missing wp_footer()
+		Functions\expect( 'did_action' )
+			->once()
+			->andReturn( 0 );
+
+		Functions\expect( 'get_transient' )
+			->once()
+			->andReturn( [] );
+
+		Functions\expect( 'get_user_meta' )
+			->once()
+			->andReturn( [] );
+
+		Functions\expect( 'get_current_user_id' )
+			->once()
+			->andReturn( 1 );
```

Comment:
You could use `Functions\when` instead of `expect` here, because we are not that interested in the calls and values returned for those functions in the test.

What we really want to validate is if `did_action` was called, and if the transient is correctly set, which we do with the above and below calls to `expect`